### PR TITLE
[ENH] l10n_th_base_location : add district and sub-district code in city.zip

### PR DESCRIPTION
**Pull request by Saran440**
- Fix duplicate name in english text file
- Add field district_code and sub_district_code on res.city.zip
- When import Thailand location, It will import district_code and sub_district_code
- if you installed module, please import location again.
- Add Test Script and data demo
- [x] Add district_code and sub_district_code in text file

### CODE REVIEW

File: l10n_th_base_location/wizard/geonames_import.py [hunk 1]
```diff
@@ -21,14 +21,44 @@ class CityZipGeonamesImport(models.TransientModel):
...
+            # There are District or Sub-District code will update code
+            if len(row) >= 6:
+                sub_district = ""
+                district = row[5]
+                if len(row) >= 7:
+                    sub_district = row[6]
+                city_zip.write(
+                    {"district_code": district, "sub_district_code": sub_district}
+                )
+        return city_zip
```
Comment (ps-tubtim) [inline l10n_th_base_location/wizard/geonames_import.py]: This code is repeated. Could you edit this?

File: l10n_th_base_location/wizard/geonames_import.py [hunk 2]
```diff
@@ -21,14 +21,44 @@ class CityZipGeonamesImport(models.TransientModel):
...
 
-    @api.depends("country_id")
+    @api.depends("country_ids")
     def _compute_is_thailand(self):
         self.ensure_one()
-        self.is_thailand = self.country_id.code == "TH"
+        if "TH" in self.country_ids.mapped("code"):
+            self.is_thailand = True
+        else:
+            self.is_thailand = False
```
Comment (ps-tubtim) [inline l10n_th_base_location/wizard/geonames_import.py]:
```suggestion
        self.is_thailand = "TH" in self.country_ids.mapped("code")
```

File: l10n_th_base_location/wizard/geonames_import.py [hunk 3]
```diff
@@ -21,14 +21,44 @@ class CityZipGeonamesImport(models.TransientModel):
         [("th", "Thai"), ("en", "English")], string="Language", default="th"
```
Comment (ps-tubtim) [inline l10n_th_base_location/wizard/geonames_import.py]:
```suggestion
        [("th", "Thai"), ("en", "English")], string="Language of Thailand", default="th"
```
I think it specifications for Thailand. Could do you think about it?

File: l10n_th_base_location/wizard/geonames_import.py [hunk 4]
```diff
@@ -18,17 +18,45 @@ class CityZipGeonamesImport(models.TransientModel):
...
+        district, sub_district = self._prepare_district_thailand(row)
+        vals.update({"district_code": district, "sub_district_code": sub_district})
+        return vals
+
+    @api.model
+    def select_zip(self, row, country, state_id):
+        city_zip = super().select_zip(row, country, state_id)
+        # Fix bug if import thailand already
+        if country.code == "TH":
+            # There are District or Sub-District code will update code
```
Comment (kittiu) [inline l10n_th_base_location/wizard/geonames_import.py]:
```suggestion
            # If District or Sub-District, update code
```

File: l10n_th_base_location/wizard/geonames_import.py [hunk 5]
```diff
@@ -18,17 +18,45 @@ class CityZipGeonamesImport(models.TransientModel):
...
+    def prepare_zip(self, row, city_id):
+        vals = super().prepare_zip(row, city_id)
+        district, sub_district = self._prepare_district_thailand(row)
+        vals.update({"district_code": district, "sub_district_code": sub_district})
+        return vals
+
+    @api.model
+    def select_zip(self, row, country, state_id):
+        city_zip = super().select_zip(row, country, state_id)
+        # Fix bug if import thailand already
```
Comment (kittiu) [inline l10n_th_base_location/wizard/geonames_import.py]:
```suggestion
```

**Review by kittiu — APPROVED** [review]

**Comment by Saran440** [comment]
@kittiu Fixed, Thank you for your reviewed.

**Review by ps-tubtim — APPROVED** [review]

**Comment by Saran440** [comment]
@sbidoul Hello, We can't merge this PR and It error about psycopg2-binary.
Could you please check something with bot?

https://travis-ci.org/github/OCA/l10n-thailand/jobs/706410531#L904

**Comment by Saran440** [comment]
@kittiu Could you try merge this again? I fix test script for import with demo data.

**Pull request merged**